src/HEAL.Attic/Core/ProtoBufSerializer.cs

```diff
@@ -11,32 +11,18 @@
 namespace HEAL.Attic {
   public sealed class ProtoBufSerializer : Serializer {
     protected override void SerializeBundle(Bundle bundle, Stream stream, bool disposeStream = true) {
-      var outputStream = new CodedOutputStream(stream);
-      try {
+      using (var outputStream = new CodedOutputStream(stream, leaveOpen: !disposeStream)) {
         bundle.WriteTo(outputStream);
-      } finally {
         outputStream.Flush();
-        if (disposeStream)
-          outputStream.Dispose();
       }
     }
 
     protected override Bundle DeserializeBundle(Stream stream, bool disposeStream = true) {
-      var inputStream = new CodedInputStream(stream);
-      try {
-        return Bundle.Parser.ParseFrom(inputStream);
-      } finally {
-        if (disposeStream)
-          inputStream.Dispose();
-      }
-    }
-
-    public override bool CanDeserialize(Stream stream) {
-      using (var inputStream = new CodedInputStream(stream)) {
+      using (var inputStream = new CodedInputStream(stream, leaveOpen: !disposeStream)) {
         try {
-          return !inputStream.IsAtEnd;
-        } catch (InvalidDataException) {
-          return false;
+          return Bundle.Parser.ParseFrom(inputStream);
+        } catch (InvalidProtocolBufferException e) {
+          throw new PersistenceException("Invalid format.", e);
```

Comment:
We should also catch InvalidDataExceptions. If someone e.g. tries to deserialize a file that has been serialized without using a DeflateStream, an InvalidDataException is thrown, because the respective Deserialize overload for files expects them to be encoded by a DeflateStream. There already exists an issue (#15) that states that all overloads should use a DeflateStream, regardless of whether (de-)serialization occurs from/to a file or a stream.

Reply:
I added that check at the abstract `Serializer`, because I think ProtoBufSerializer shouldn't care about DeflateStream.